python/ray/_private/accelerators/nvidia_gpu.py

```diff
@@ -49,9 +54,44 @@ def get_current_node_num_accelerators() -> int:
             pynvml.nvmlInit()
         except pynvml.NVMLError:
             return 0  # pynvml init failed
+        driver_version = pynvml.nvmlSystemGetDriverVersion()
         device_count = pynvml.nvmlDeviceGetCount()
+        cuda_devices = []
+        for index in range(device_count):
+            handle = pynvml.nvmlDeviceGetHandleByIndex(index)
```

Comment:
You may get `NVMLError_GpuIsLost` here if the GPU is in an invalid state.

```suggestion
            try:
                handle = pynvml.nvmlDeviceGetHandleByIndex(index)
            except pynvml.NVMLError_GpuIsLost:
                continue
```

Reply:
thanks for catching that out